src/machine.rs

```diff
@@ -591,12 +601,26 @@ impl<'mir, 'tcx> MiriMachine<'mir, 'tcx> {
 
 impl VisitMachineValues for MiriMachine<'_, '_> {
     fn visit_machine_values(&self, visit: &mut impl FnMut(&Operand<Provenance>)) {
-        // FIXME: visit the missing fields: env vars, weak mem, the MemPlace fields in the machine,
-        // DirHandler, extern_statics, the Stacked Borrows base pointers; maybe more.
-        let MiriMachine { threads, tls, .. } = self;
+        // FIXME: visit the missing fields: weak mem, DirHandler, Stacked Borrows base pointers(?); maybe more.
+        let MiriMachine { threads, tls, env_vars, argc, argv, cmd_line, extern_statics, .. } = self;
 
         threads.visit_machine_values(visit);
         tls.visit_machine_values(visit);
+        env_vars.visit_machine_values(visit);
+
+        if let Some(argc) = argc {
+            visit(&Operand::Indirect(*argc));
+        }
+        if let Some(argv) = argv {
+            visit(&Operand::Indirect(*argv));
+        }
+        if let Some(cmd_line) = cmd_line {
+            visit(&Operand::Indirect(*cmd_line));
+        }
+        for ptr in extern_statics.values().copied() {
+            let ptr: Pointer<Option<Provenance>> = ptr.into();
+            visit(&Operand::Indirect(MemPlace::from_ptr(ptr)));
```

Comment:
This is pretty ugly. Is there a better way to do this? Or should we adjust the interface of the visitor so we can just pass a pointer to it directly?

Reply:
We could have more `From` instances to convert pointers to MemPlaces or so... :shrug: